Lib/test/test_pyexpat.py

```diff
@@ -821,5 +823,169 @@ def start_element(name, _):
         self.assertEqual(started, ['doc'])
 
 
+class AttackProtectionTestCases:
```

Comment:
```suggestion
class AttackProtectionTestBase(ABC):
```
ideas:
- singular
- communicate abstract base class nature better (in particular also because it feels to much like a mix-in than it is)
- allow ABC to do additional checks

Reply:
We have a bit of TestCase and TestBase in the code base and we usually don't use ABCs, but considering the nature of the tests, I think it's fine to use this. I personally prefer TestCase because it's only the cases. TestBase feels to me something that is even higher level (just the interface `set_*` for instance).